### ISSUE TITLE
Fix accessibility and interaction of the Video block Text tracks popover

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66717]
### Description

The UI to add Text tracks to a Video block lives in a Popover that opens from the block toolbar.

This UI has 3 different states with a different UI. All of them render within an ARIA role=menu, which is provides specific semantic and keyboard interaction which are appropriate for only one of the states. The other two need to be refactored to provide better accessibility and improve hte expected interaction.

It is worth reminding, as reported many times already, that ARIA menus can only contain menu items, their variants manu item radio / checkbox, groups and separators.
The expected keyboard interaction is to be able to narivate the menu items with the arrow keys.

## Initial state

![Image](https://github.com/user-attachments/assets/84b1bc08-1c86-48a7-8bb6-3976a3d79f0f)

The ARIA menu contains a descriptive paragraph. At the very least, this paragraph should be moved outiside of the ARIA menu. 
Howeer, given the expected keyboard interaction, screen reader users may not have a clue some extra content is rendered in this popover because they will likely use arrow keys navigation to explore the menu thus entirely skipping the pargraph. 
I'd argue the paragraph should be enitrely removeed as a dropdown with an ARIA menu isn;t the right place for descriptions or instructions.

## State with list of uploaded tracks

![Image](https://github.com/user-attachments/assets/4914f1fa-33db-43b9-b9bf-bdb9dcc0f3db)

This is the only state where semantics, ARIA roles and interaction are the expected ones.
The menu only contains menu items, split in two groups. Keyboard interaction works as expected. All content is perceivable.

## Edit state 

![Image](https://github.com/user-attachments/assets/c59e7d52-f910-4c90-b7e7-7f3267c4ebfd)

In this state, the ARIA menu contains a form elements like inputs and buttons. This is incorrect. Arrow keys navigation still kicks in and moves focus through the focusable elements, which is unexpected. It also fails on the select element, as on macOS the down arrow key opens the select. This UI is basically a form and should not be inside an ARIA menu.
In this state, the UI should switch to a generic Popover content with no ARIA menu.

In similar cases, for example the Replace media dropdown, the ARIA menu only contains menu items. The additional link editing UI is a form that is rendered outisde of the menu. Screenshot:

![Image](https://github.com/user-attachments/assets/63a6f1ef-4223-43f7-8573-b49312f70a71)

Still not ideal, but it's a better pattern.

### Step-by-step reproduction instructions

- Add a Video block
- Click 'Text tracks' in the block toolbar.
- Observe the Popover initial UI.
- Add some tracks (.vtt files) either from the Media Library or by direct upload.
- Observe the editing UI form elements still live within an ARIA menu (inspect the source).
- Click 'Close' and observe the UI listing the uploaded tracks.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by afercia** [comment]
Noticed that when the track editor UI is switched to the 'edit' view _and_ closing the Popover by pressing the Escape key, the track editor stays in 'editing' mode. In fact, opening it again shows the editing UI. This may be confusing for users. I think the intiial view should always be the track list.
Animated GIF to illustrate with focus highlighted with a red outline:

![Image](https://github.com/user-attachments/assets/585b43b8-a027-45fb-a95c-76d8b7f133b0)
